src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -66,7 +66,13 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
 
   /** Ensure that the internal array has at least `n` cells. */
   protected def ensureSize(n: Int): Unit = {
-    array = ArrayBuffer.ensureSize(array, size0, n)
+    array = ArrayBuffer.ensureSize(array, size0, n.toLong)
+  }
+
+  // TODO 3.T: should be `protected`, perhaps `protected[this]`
+  /** Ensure that the internal array has at least `n` additional cells more than `size0`. */
+  private[mutable] def ensureAdditionalSize(n: Int): Unit = {
+    array = ArrayBuffer.ensureSize(array, size0, size0.toLong + n)
```

Comment:
Here .toLong is necessary. Perhaps a comment might be added to explain why? I am not sure if it is clean enough from the code.

Reply:
if you don't do `toLong` on one of them first, it adds the two `Int`s together, which has the same overflow problem. to me, at least, that's obvious? it's fairly common to call `to<number_type>` on one argument before doing maths, to ensure that the operation is done using the correct type (e.g. calling `toDouble` to ensure you get floating point division rather than integer division).